src/test/java/com/wikia/webdriver/common/testnglisteners/BrowserAndTestEventListener.java

```diff
@@ -0,0 +1,265 @@
+import com.wikia.webdriver.common.core.configuration.Configuration;
+import com.wikia.webdriver.common.core.elemnt.JavascriptActions;
+import com.wikia.webdriver.common.core.helpers.User;
+import com.wikia.webdriver.common.core.networktrafficinterceptor.NetworkTrafficInterceptor;
+import com.wikia.webdriver.common.driverprovider.DriverProvider;
+import com.wikia.webdriver.common.logging.Log;
+import com.wikia.webdriver.common.logging.VelocityWrapper;
+import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+
+import org.joda.time.DateTime;
+import org.openqa.selenium.By;
+import org.openqa.selenium.Cookie;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebDriverException;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.events.AbstractWebDriverEventListener;
+import org.testng.ITestContext;
+import org.testng.ITestListener;
+import org.testng.ITestResult;
+import org.testng.SkipException;
+
+import java.lang.reflect.Method;
+import java.util.Date;
+
+public class BrowserAndTestEventListener extends AbstractWebDriverEventListener
+    implements ITestListener {
+
+  private By lastFindBy;
+  private WikiaWebDriver driver;
+
+  @Override
+  public void beforeNavigateTo(String url, WebDriver driver) {
+    new JavascriptActions(driver).execute("window.stop()");
+
+    Log.ok("Navigate to", VelocityWrapper.fillLink(url, url));
+    Log.logJSError();
+  }
+
+  @Override
+  public void afterNavigateTo(String url, WebDriver driver) {
+    Method method = TestContext.getCurrentTestMethod();
+    Class<?> declaringClass = method.getDeclaringClass();
+
+    String cookieDomain = String.format(".%s", Configuration.getEnvType().getWikiaDomain());
+
+    Date cookieDate = new Date(new DateTime().plusYears(10).getMillis());
+
+    if (!AlertHandler.isAlertPresent(driver)) {
+      String command = "Url after navigation";
+      if (url.equals(driver.getCurrentUrl())) {
+        Log.ok(command, VelocityWrapper.fillLink(driver.getCurrentUrl(), driver.getCurrentUrl()));
+      } else {
+        if (driver.getCurrentUrl().contains("data:text/html,chromewebdata ")) {
+          driver.get(url);
+          Log.warning(command, driver.getCurrentUrl());
+        } else if (driver.getCurrentUrl().contains(URLsContent.NOT_A_VALID_COMMUNITY)) {
+          Log.warning(command, driver.getCurrentUrl());
+          driver.get("http://seleniumworkaround");
+          Log.log("Workaround", "Dummy redirect", true);
+        } else {
+          Log.warning(command, driver.getCurrentUrl());
+        }
+      }
+    } else {
+      Log.warning("Url after navigation", "Unable to check URL after navigation - alert present");
+    }
+
+    if (driver.getCurrentUrl().contains(Configuration.getWikiaDomain())) {
+      // HACK FOR DISABLING NOTIFICATIONS
+      try {
+        new JavascriptActions(driver).execute("$(\".sprite.close-notification\")[0].click()");
+      } catch (WebDriverException e) {
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1166)